src/tls/incoming.rs

```diff
@@ -158,6 +162,25 @@ impl MaybeTlsIncomingStream<TcpStream> {
         Ok(())
     }
 
+    #[cfg(feature = "sources-utils-tcp-keepalive")]
+    pub(crate) fn set_keepalive(
+        &mut self,
+        keepalive: Option<TcpKeepaliveConfig>,
+    ) -> io::Result<()> {
```

Comment:
Since the underlying function takes the value directly and not an `Option`, would it make sense to have this take `TcpKeepaliveConfig` directly and move the `if let Some` into the callers? There appears to only be one caller, so it shouldn't make much difference.

Reply:
Oh, that's a good point. I was fixating on https://docs.rs/tokio/0.2.23/tokio/net/struct.TcpStream.html#method.set_keepalive taking an `Option`, but it only relates to the `time`, which is also an `Option` in our struct.

Will address.